libs/ledger-live-common/src/families/icon/transaction.ts

```diff
@@ -0,0 +1,58 @@
+import type { Transaction, TransactionRaw } from "./types";
+import { BigNumber } from "bignumber.js";
+import {
+  formatTransactionStatusCommon as formatTransactionStatus,
+  fromTransactionCommonRaw,
+  fromTransactionStatusRawCommon as fromTransactionStatusRaw,
+  toTransactionCommonRaw,
+  toTransactionStatusRawCommon as toTransactionStatusRaw,
+} from "@ledgerhq/coin-framework/transaction/common";
+import type { Account } from "@ledgerhq/types-live";
+import { getAccountUnit } from "../../account";
+import { formatCurrencyUnit } from "../../currencies";
+
+export const formatTransaction = (
+  { mode, amount, recipient, useAllAmount }: Transaction,
+  account: Account,
+): string =>
+  `
+${mode.toUpperCase()} ${
+    useAllAmount
+      ? "MAX"
+      : amount.isZero()
+      ? ""
+      : " " +
+        formatCurrencyUnit(getAccountUnit(account), amount, {
+          showCode: true,
+          disableRounding: true,
+        })
+  }${recipient ? `\nTO ${recipient}` : ""}`;
+
+export const fromTransactionRaw = (tr: TransactionRaw): Transaction => {
+  const common = fromTransactionCommonRaw(tr);
+  return {
+    ...common,
+    family: tr.family,
+    mode: tr.mode,
+    fees: tr.fees ? new BigNumber(tr.fees) : null,
+  };
+};
+
+export const toTransactionRaw = (t: Transaction): TransactionRaw => {
+  const common = toTransactionCommonRaw(t);
+  return {
+    ...common,
+    family: t.family,
+    mode: t.mode,
+    fees: t.fees?.toString() || null,
+  };
+};
```

Comment:
Need to also handle `stepLimit` here

Reply:
@heirland

Reply:
@mangcanbang
- in `fromTransactionRaw` you need to convert `tr.stepLimit` to `Transaction.stepLimit` type
- in `toTransactionRaw` you need to convert from `t.stepLimit` to `TransactionRaw.stepLimit` type